pkg/kubeapiserver/authenticator/config.go

```diff
@@ -55,7 +55,7 @@ import (
 
 // Config contains the data on how to authenticate a request to the Kube API Server
 type Config struct {
-	Anonymous      bool
+	Anonymous      *apiserver.AnonymousAuthConfig
```

Comment:
I *think* this field should go away in favor of the field inside `AuthenticationConfig` to avoid duplicate information

The trick will be when the user specified `--anonymous-auth` and `--authentication-config` which is allowed today... I guess we'd have to pull in the config file they specified, make sure it didn't set anonymous fields, then populate the config from the anonymous flag they specified? @enj @aramase, how were you envisioning moving more pieces of authn config from flags into the config file?

Reply:
for reference, when OIDC config was able to be specified in the config file, the OIDC fields in this struct went away in favor of `AuthenticationConfig *apiserver.AuthenticationConfiguration`, and if oidc flags were specified, a synthetic authentication config was constructed:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubeapiserver/options/authentication.go#L473-L541

Reply:
Done.